File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/data/type/XmlValueType.ts
module api.data.type {

    export class XmlValueType extends ValueType {

        constructor() {
            super("Xml");
        }

        isValid(value: any): boolean {
            return typeof value === 'string';
        }

        valueEquals(a: string, b: string): boolean {
            return api.ObjectHelper.stringEquals(a, b);
        }
    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/data/XmlValueType.java
package com.enonic.xp.data;

final class XmlValueType
    extends ValueType<String>
{
    XmlValueType()
    {
        super( "Xml", JavaTypeConverters.STRING );
    }

    @Override
    public Value fromJsonValue( final Object object )
    {
        return ValueFactory.newXml( convertNullSafe( object ) );
    }
}
